sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatAsyncClient.java

```diff
@@ -0,0 +1,243 @@
+        this.serviceClient.listChatThreadsNext(nextLink, callback);
+    }
+
+    /**
+     * A builder for creating a new instance of the ChatAsyncClient type.
+     */
+    public static final class Builder {
+        /*
+         * The endpoint of the Azure Communication resource.
+         */
+        private String endpoint;
+
+        /**
+         * Sets The endpoint of the Azure Communication resource.
+         *
+         * @param endpoint the endpoint value.
+         * @return the Builder.
+         */
+        public Builder endpoint(String endpoint) {
+            this.endpoint = endpoint;
+            return this;
+        }
+
+        /*
+         * The Azure Core generic ServiceClient Builder.
+         */
+        private ServiceClient.Builder serviceClientBuilder;
+
+        /**
+         * Sets The Azure Core generic ServiceClient Builder.
+         *
+         * @param serviceClientBuilder the serviceClientBuilder value.
+         * @return the Builder.
+         */
+        public Builder serviceClientBuilder(ServiceClient.Builder serviceClientBuilder) {
+            this.serviceClientBuilder = serviceClientBuilder;
+            return this;
+        }
+
+        /*
+         * The Interceptor to set intercept request and set credentials.
+         */
+        private Interceptor credentialInterceptor;
+
+        private Interceptor userAgentInterceptor;
+
+        /**
+         * Sets The Interceptor to set intercept request and set credentials.
+         *
+         * @param credentialInterceptor the credentialInterceptor value.
+         * @return the Builder.
+         */
+        public Builder credentialInterceptor(Interceptor credentialInterceptor) {
+            this.credentialInterceptor = credentialInterceptor;
+            return this;
+        }
+
+        public Builder userAgentInterceptor(Interceptor userAgentInterceptor) {
+            this.userAgentInterceptor = userAgentInterceptor;
+            return this;
+        }
+
+        /**
+         * Builds an instance of ChatAsyncClient with the provided parameters.
+         *
+         * @return an instance of ChatAsyncClient.
+         */
+        public ChatAsyncClient build() {
+            if (serviceClientBuilder == null) {
+                if (endpoint == null) {
+                    throw new IllegalArgumentException("Missing required parameters 'endpoint'.");
+                }
+                this.serviceClientBuilder = new ServiceClient.Builder();
+            }
+            if (endpoint != null) {
+                final String retrofitBaseUrl = endpoint;
+                serviceClientBuilder.setBaseUrl(retrofitBaseUrl);
+            }
+            if (credentialInterceptor != null) {
+                serviceClientBuilder.setCredentialsInterceptor(credentialInterceptor);
+            }
+            if (userAgentInterceptor != null) {
+                serviceClientBuilder.addInterceptor(userAgentInterceptor);
+            }
```

Comment:
I thought we'd add a default agent interceptor here, c.f. your other PR. Do you plan to add it later?

Reply:
It seems involve a large refactor to four ChatClient files according to comment in that PR, I would not include that part in preview 3.